jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TaskCommon.java

```diff
@@ -17,25 +17,58 @@
 package com.google.cloud.tools.jib.gradle;
 
 import com.google.api.client.http.HttpTransport;
+import com.google.cloud.tools.jib.ProjectInfo;
 import com.google.cloud.tools.jib.api.AbsoluteUnixPath;
 import com.google.cloud.tools.jib.api.FilePermissions;
+import com.google.cloud.tools.jib.api.LogEvent;
+import com.google.cloud.tools.jib.plugins.common.ProjectProperties;
+import com.google.cloud.tools.jib.plugins.common.UpdateChecker;
+import com.google.common.util.concurrent.Futures;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 import java.util.logging.Level;
 import javax.annotation.Nullable;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
 import org.gradle.api.UnknownTaskException;
+import org.gradle.api.logging.Logger;
 import org.gradle.api.plugins.WarPlugin;
 import org.gradle.api.tasks.TaskProvider;
-import org.gradle.internal.logging.events.LogEvent;
 import org.gradle.internal.logging.events.OutputEventListener;
 import org.gradle.internal.logging.slf4j.OutputEventListenerBackedLoggerContext;
 import org.slf4j.LoggerFactory;
 
 /** Collection of common methods to share between Gradle tasks. */
 class TaskCommon {
 
+  public static final String VERSION_URL = "https://storage.googleapis.com/jib-versions/jib-gradle";
+
+  static Future<Optional<String>> newUpdateChecker(
+      ProjectProperties projectProperties, Logger logger) {
+    if (projectProperties.isOffline() || !logger.isLifecycleEnabled()) {
+      return Futures.immediateFuture(Optional.empty());
+    }
+    return UpdateChecker.checkForUpdate(
+        Executors.newSingleThreadExecutor(), projectProperties::log, VERSION_URL);
```

Comment:
If `newUpdateChecker` creates `Executors.newSingleThreadExecutor()`, let's make it "close" it, i.e., calling `shutdown()` in `newUpdateChecker`. And since I know our plugins just terminate without even calling `shutdown()`, but it doesn't harm to guarantee it in `finally`.
```java
} finally {
   executorService.shutdown();
}
```